### ISSUE TITLE
[device-port][halium-7.1][gtexswifi] Galaxy Tab A 7.0

### ISSUE THREAD

**Issue by bfourk** [Halium/projectmanagement#308]
Tree: halium-7.1

- [x] Create manifest https://github.com/bfourk/halium-devices/blob/halium-7.1/manifests/samsung_gtexswifi.xml
- [x] Boot image and system image build successfully
- [x] Device boots into rootfs, `usb: Manufacturer: GNU/Linux Device` appears in `dmesg` on host.
- [ ] LXC container starts and does not crash
- [ ] libhybris tests
  * [ ] test_gps
  * [ ] test_hwcomposer
  * [ ] test_lights
  * [ ] test_vibrator
  * [ ] test_wifi
  * [ ] test_sensors
  * [ ] test_audio
  * [ ] test_camera
  * [ ] test_input
  * [ ] test_nfc
  * [ ] test_recorder

**NOTE: this tablet requires a 512-byte header appended to the boot.img before it will even attempt to boot. This header is the letters "DHTB" followed by 508 NULL bytes.**

**Comment by bfourk** [comment]
Managed to patch the kernel and it compiles with CONFIG_TMPFS_POSIX_ACL set, haven't attempted to boot it yet.

**Comment by bfourk** [comment]
I've gotten as far as booting into the init environment, and telnet'ing into the tablet. I haven't managed to get it to boot into the rootfs, not with hybris-boot or halium-boot.

**Comment by 19atlas** [comment]
halium is booting?
because [documation says](https://docs.halium.org/en/latest/porting/build-sources.html#modify-the-kernel-configuration) CONFIG_IKCONFIG and CONFIG_IKCONFIG_PROC must be y

**Comment by bfourk** [comment]
It's been a while since I've worked on this, but I was attempting to port Ubuntu Touch, and I got as far as the startup screen showing and freezing. I could still use ssh via. USB internet to control the tablet and it appeared to be booted into the distro (I believe I used halium-boot).

I used the kernel config check to set the configuration, and I remember having to unset CONFIG_USER_NS because it was causing compile problems.

**Comment by 19atlas** [comment]
Well, if there are any files you use (such as defconfig), could you share them? 

also gtexswifi needs DHTB to booting ubuntu touch?

(booting is important to me to port postmarketos)

**Comment by bfourk** [comment]
Yes, the tablet will refuse to use the boot.img unless the DHTB part is appended to the beginning of the image (at least on mine)

I have tried porting pmOS as well, and ran into much the same issue as Ubuntu Touch (graphics froze on startup screen, could still SSH into the tablet)

I unfortunately don't have the defconfig anymore for Ubuntu Touch, but I do have the defconfig+files from pmOS, I've attached them to this comment.

[pmos-gtexswifi.zip](https://github.com/user-attachments/files/16014649/pmos-gtexswifi.zip)

**Comment by 19atlas** [comment]
> Yes, the tablet will refuse to use the boot.img unless the DHTB part is appended to the beginning of the image (at least on mine)
> 
> I have tried porting pmOS as well, and ran into much the same issue as Ubuntu Touch (graphics froze on startup screen, could still SSH into the tablet)
> 
> I unfortunately don't have the defconfig anymore for Ubuntu Touch, but I do have the defconfig+files from pmOS, I've attached them to this comment.
> 
> [pmos-gtexswifi.zip](https://github.com/user-attachments/files/16014649/pmos-gtexswifi.zip)

thank you so much
btw "refuse" do you mean to stuck on Samsung logo (my device do this)?
also I couldn't find anything about DHTB in the deviceinfo you shared. [could i use this?](https://github.com/osm0sis/dhtbsign)

**Comment by bfourk** [comment]
Yes, it would get stuck on the Samsung logo without the header.

So, for pmOS I would export the boot.img, and before flashing it to my tablet, I wrote a script that would add the header (for some reason the tablet just checks for the headers existence, it doesn't check if it's valid) You know it's working if there's red text on the top of the screen a few seconds after the Samsung logo pops up

I wrote a small bash script to do this for me, you can use it `./append-header boot.img` and it will automatically append the header to the beginning of the image for you. It's only 3 lines, and you can use a hex editor if you want instead (just copy the 512-byte file to the beginning of the image file)

[append-header.zip](https://github.com/user-attachments/files/16015716/append-header.zip)

**Comment by 19atlas** [comment]
> I wrote a small bash script to do this for me, you can use it `./append-header boot.img` and it will automatically append the header to the beginning of the image for you. It's only 3 lines, and you can use a hex editor if you want instead (just copy the 512-byte file to the beginning of the image file)
> 
> [append-header.zip](https://github.com/user-attachments/files/16015716/append-header.zip)

finaly i can see postmarketOS (thank you again). Could I use the script you sent in my own port in the future? (also this script maybe work gtexlte[sm-t285])

**Comment by bfourk** [comment]
Yes, you can use the script. I don't have gtexslte, so I'm not sure if the header will work there or if it needs to be valid (my script just appends a null header which is enough for gtexswifi)
